### PR TITLE
Minimize LBRY app to tray.

### PR DESCRIPTION
**Pull request by hackrush01**

### CODE REVIEW

**Comment by kauffj** [comment]
@hackrush01 this has a significant bug: closing the window will close the daemon, even though the app stays open. I believe this will require changes to how `before-quit` and or `window-all-closed` is currently being handled.

I also changed the behavior so that only close moves to tray and that minimize is just minimize.

**Comment by kauffj** [comment]
@hackrush01 at least on Linux, I get an error now every time I close and re-open the app. Please test in Ubuntu VM if you are not seeing this behavior.

Additionally, I do not want "Minimize if closed" to be a context menu in the tray. We can consider adding this in settings, but I don't think this option is necessary currently (and if we add an option for minimize, we should also turn close behavior into an option between tray or quit).

**Comment by kauffj** [comment]
https://spee.ch/8/tray-error.png

**Comment by tzarebczan** [comment]
I've seen that as a context menu setting on other Windows apps I've used, but I also agree that the settings would make more sense. Users with BW caps most likely want to close as a default behavior.

**Comment by hackrush01** [comment]
Known bug: error on start. Working on it

**Comment by hackrush01** [comment]
@kauff I've added `Alt+F4` as an accelerator to quit the app. Though I suggest using a different combination(such as `Alt+X` or other). As we could want the default actions for all default quitting techniques to minimize the app to tray, and a special, different action for quitting the app.

File: app/main.js [hunk 1]
```diff
@@ -167,17 +174,84 @@ function createWindow () {
...
+  return [
+    getToggleItem(),
+    {
+      label: "Quit",
+      click: () => safeQuit(),
+    },
+  ]
+
+  function getToggleItem () {
+    if (win.isVisible()) {
```
Comment (kauffj) [inline app/main.js]: I think this needs to be a check if the window is focused, not just if it is visible. If I Alt+Tab from an open app, I am shown "Hide" in the context menu, when it should be "Show".

File: app/main.js [hunk 2]
```diff
@@ -167,17 +174,84 @@ function createWindow () {
...
+  win.on("show", () => {
+    if (minimize) updateTray();
+  });
+
+  win.on("blur", () => {
+    globalShortcut.unregister('Alt+F4');
+  });
+
+  win.on("focus", () => {
+    globalShortcut.register('Alt+F4', () => safeQuit());
```
Comment (kauffj) [inline app/main.js]: This should be Ctrl/Cmd + Q as well.

Comment (kauffj) [reply app/main.js]: (Or if possible, detect OS setting for this, if one exists)

File: app/main.js [hunk 3]
```diff
@@ -167,17 +174,84 @@ function createWindow () {
...
+  });
+
   // Menu bar
   win.setAutoHideMenuBar(true);
   win.setMenuBarVisibility(isDebug);
   setMenu();
 
 };
 
+function createTray () {
```
Comment (kauffj) [inline app/main.js]: Can you add a behavior that double clicking the tray icon also triggers the show behavior?

File: app/main.js (hunk 2 above)
Comment (kauffj) [reply app/main.js]: Also needs to be available via a File menu in the app, I think.

**Comment by tzarebczan** [comment]
@kauffj - running LBRY again when its minimized to tray does not bring the app to the foreground. Neither does clicking lbry:// hyperlinks, but the link is opened when the app is manually shown again.  This was on Windows, not sure about other OS but I can try later if needed. 

When LBRY is running in a maximized window - I also noticed that if you minimize LBRY, then close the app (right click taskbar icon > close) and finally bring it back from the tray, the Window size goes back to the previous unmaximized size.

**Comment by kauffj** [comment]
@hackrush01 at a minimum the app will need to be brought to the foreground when the URI handler is initiated. Ideally, it'd come to the foreground whenever triggered. And the sizing issue ought to be fixed as well.

File: app/main.js (hunk 2 above)
Comment (hackrush01) [reply app/main.js]: There is a predefined accelerator for this `CommandOrControl+Q`, which works on all OSes

And regarding the File Menu in the App, I already gave it a shot, and ran into some issues, I'll work on it. I would probably need to look at `module.exports`, as i need to "export" the `safeQuit()` function from `main.js` to `menu/main-menu.js`

File: app/main.js (hunk 3 above)
Comment (hackrush01) [reply app/main.js]: Done.

File: app/main.js (hunk 1 above)
Comment (hackrush01) [reply app/main.js]: Done

**Comment by hackrush01** [comment]
@tzarebczan @kauffj I think the other 2 issues are also solved. I could test it as clicking links doesn't open the app in my laptop, and I couldn't reproduce the maximization issue.

**Comment by hackrush01** [comment]
Update: @kauffj I added the quit option to file menu in the menu bar too :-)

**Comment by tzarebczan** [comment]
Hyperlinks bring the app to focus now from the tray, woohoo. 

Launching LBRY again while it's in the tray does not bring it up. 

If the LBRY app is not maximized and closed to tray, it comes back maximized. 

When the app is in focus, the right click still show "show app" - should it say hide app and have the same function as close?

**Pull request merged**